src/one/nio/http/QueryParameterIterator.java

```diff
@@ -0,0 +1,127 @@
+package one.nio.http;
+
+import one.nio.util.URLEncoder;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * {@link Iterator} over {@code String} {@code key[=[value]]} query parameters in order skipping empty parameters
+ *
+ * @author incubos
+ */
+final class QueryParameterIterator implements Iterator<Map.Entry<String, String>> {
+    private final String query;
+    private int cur; // Always points at the current parameter or -1 (EOF)
+
+    QueryParameterIterator(String query) {
+        assert query != null;
+
+        this.query = query;
+        this.cur = 0;
+
+        advance();
+    }
+
+    @Override
+    public boolean hasNext() {
+        return cur >= 0;
+    }
+
+    @Override
+    public Map.Entry<String, String> next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException("Beyond query parameters");
+        }
+
+        // Extract parameter
+        final int next = query.indexOf('&', cur);
+        final String parameter;
+        if (next == -1) {
+            // Last parameter
+            parameter = query.substring(cur);
+            cur = -1;
+        } else {
+            // Advance
+            parameter = query.substring(cur, next);
+            cur = next;
+            advance();
+        }
+
+        assert !parameter.isEmpty();
+
+        return parse(parameter);
+    }
+
+    /**
+     * Skip '&'s and position at the next parameter or stops
+     */
```

Comment:
This is not a javadoc. Please convert to a regular comment.
The same applies to all other comments to private members.

Reply:
Fixed.